src/client/app/containers/view.mode.ts

```diff
@@ -27,43 +29,54 @@ import { Strings } from '../strings';
 export class ViewMode implements OnInit, OnDestroy {
     strings = Strings();
     snippet: ISnippet;
-    showInfo: boolean;
     paramsSub: Subscription;
 
     constructor(
         private _store: Store<fromRoot.State>,
+        private _request: Request,
         private _route: ActivatedRoute
     ) {
         this._store.select(fromRoot.getCurrent).subscribe(snippet => {
             this.snippet = snippet;
         });
     }
 
+    get host() {
+        return environment.current.host.toLowerCase();
+    }
+
     ngOnInit() {
         this.paramsSub = this._route.params
-            .subscribe(async(params) => {
-                if (environment.current.host.toUpperCase() !== params.host.toUpperCase()) {
-                    environment.current.host = params.host.toUpperCase();
+            .map(params => ({ type: params.type, host: params.host, id: params.id }))
+            .mergeMap(({ type, host, id }) => {
+                if (environment.current.host.toUpperCase() !== host.toUpperCase()) {
+                    environment.current.host = host.toUpperCase();
                     // Update environment in cache
                     environment.current = environment.current;
-                    await applyTheme(environment.current.host);
                 }
 
-                let urlSegments = this._route.snapshot.url;
-                switch (urlSegments[1].path) {
+                switch (type) {
                     case 'private-samples':
```

Comment:
We still need some kind of differentiation in the path, to distinguish between samples and gists. We could technically have just a 'sample' vs 'gist' (because we now have the all samples JSON file so only need the ID) but I kept the private-samples vs sample for the purposes of explicit differentiation.